tests/lifecycle_wait_test.go

```diff
@@ -126,21 +125,24 @@ func TestLifecycleWaitForNavigation(t *testing.T) {
 			pingSlowness: 0,
 			pingJSSlow:   false,
 			waitUntil:    common.LifecycleEventNetworkIdle,
-			assertFunc: func(tb *testBrowser, p api.Page) testPromise {
+			assertFunc: func(tb *testBrowser, p api.Page) error {
 				result := p.TextContent("#pingRequestText", nil)
 				assert.EqualValues(t, "Waiting... pong 10 - for loop complete", result)
 
-				opts := tb.toGojaValue(&common.FrameWaitForNavigationOptions{
-					Timeout:   1000,
-					WaitUntil: common.LifecycleEventNetworkIdle,
-				})
-				waitForNav := k6ext.Promise(tb.vu.Context(), func() (result any, reason error) {
-					return p.WaitForNavigation(opts)
-				})
+				waitForNav := func() error {
+					opts := tb.toGojaValue(&common.FrameWaitForNavigationOptions{
+						Timeout:   1000,
+						WaitUntil: common.LifecycleEventNetworkIdle,
+					})
+					_, err := p.WaitForNavigation(opts)
+					return err
+				}
 
-				return tb.promise(waitForNav)
+				ctx, cancel := context.WithTimeout(tb.ctx, 5*time.Second)
```

Comment:
Should the following be like:

```suggestion
				ctx, cancel := context.WithTimeout(tb.ctx, time.Second)
```

As the error message on line 145 says: _"waiting for navigation: timed out after **1s**"._

Reply:
The timeout test runs an asserFunc instead of running the main body of the test, so it completes before reaching the 5 second timeout step.